File: src/main/java/me/aov/sellgui/managers/ItemNBTManager.java
package me.aov.sellgui.managers;

import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.bukkit.persistence.PersistentDataType;

public interface ItemNBTManager {
    double getSellPrice(ItemStack item);
    void setSellPrice(ItemStack item, double price);
    boolean needsEvaluation(ItemStack item);
    void setNeedsEvaluation(ItemStack item, boolean needsEvaluation);
    String getItemName(ItemStack item);

    // New methods for NBT tag manipulation
    void addNBTTag(ItemStack itemStack, String key, String value);
    String getNBTTag(ItemStack itemStack, String key); // Removed PersistentDataType parameter
}

File: src/main/kotlin/me/aov/sellgui/managers/ItemNBTManager.kt
package me.aov.sellgui.managers

import org.bukkit.inventory.ItemStack

interface ItemNBTManager {
    fun getSellPrice(item: ItemStack?): Double

    fun setSellPrice(item: ItemStack?, price: Double)

    fun needsEvaluation(item: ItemStack?): Boolean

    fun setNeedsEvaluation(item: ItemStack?, needsEvaluation: Boolean)

    fun getItemName(item: ItemStack?): String

    fun addNBTTag(itemStack: ItemStack?, key: String, value: String)

    fun getNBTTag(itemStack: ItemStack?, key: String): String?
}
